src/overlays/actors/ovl_Demo_Tre_Lgt/z_demo_tre_lgt.c

```diff
@@ -23,21 +35,136 @@ const ActorInit Demo_Tre_Lgt_InitVars = {
     (ActorFunc)DemoTreLgt_Update,
     (ActorFunc)DemoTreLgt_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Tre_Lgt/DemoTreLgt_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Tre_Lgt/DemoTreLgt_Destroy.s")
+static TransformUpdateIndex* sTransformUpdIdx[] = { 0x06004B60, 0x06004F70 };
+
+static DemoTreLgtActionFunc sActionFuncs[] = {
+    func_8099375C,
+    func_80993848,
+};
+
+void DemoTreLgt_Init(Actor* thisx, GlobalContext* globalCtx) {
+    DemoTreLgt* this = THIS;
+
+    if (!SkelCurve_Init(globalCtx, &this->skelCurve, &D_06005EB8, sTransformUpdIdx[0])) {
+        osSyncPrintf("Demo_Tre_Lgt_Actor_ct();コンストラクト失敗\n");
+    }
+
+    // defined in .rodata but not used
+    "1";
+    "../z_demo_tre_lgt.c";
+
+    this->unk_170 = 255;
+    this->unk_174 = 255;
+    this->status = 0;
+    func_80993754(this);
+}
+
+void DemoTreLgt_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    DemoTreLgt* this = THIS;
+
+    SkelCurve_Destroy(globalCtx, &this->skelCurve);
+}
+
+void func_80993754(DemoTreLgt* this) {
+    this->action = DEMO_TRE_LGT_ACTION_WAIT;
+}
+
+void func_8099375C(DemoTreLgt* this, GlobalContext* globalCtx) {
+    EnBox* treasureChest = (EnBox*)this->actor.parent;
+
+    if (treasureChest != NULL && func_800A56C8(&treasureChest->skelanime, 10.0f)) {
+        func_809937B4(this, globalCtx, treasureChest->skelanime.animCurrentFrame);
+    }
+}
+
+void func_809937B4(DemoTreLgt* this, GlobalContext* globalCtx, f32 currentFrame) {
+    SkelAnimeCurve* skelCurve = &this->skelCurve;
+    s32 pad;
+
+    this->action = DEMO_TRE_LGT_ACTION_ANIMATE;
+
+    { s32 pad; }
```

Comment:
Is a separate pad in a block really needed here? Ideally it should just be `s32 pad[2]` at the top of the function, if it can be.

Alternatively, it might be worth looking into using temps for `sDemoTreLgtInfo.frameInfo[gSaveContext.linkAge]` and/or using `((void)0, gSaveContext.linkAge)`

Reply:
Everything gets messed up when using temps or the gSaveContext thing. the only thing that works is the pad, and yeah pad[2] works